corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -134,19 +135,35 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
     };
 
     FormplayerFrontend.getChannel().reply("app:select:menus", function (options) {
-        var route = "navigate_menu";
-        if (options.isInitial) {
-            route = "navigate_menu_start";
-        } else if (options.endpointId) {
-            route = "get_endpoint";
+        if (!options.endpointId) {
+            return API.queryFormplayer(options, options.isInitial ? "navigate_menu_start" : "navigate_menu");
         }
 
-        var deferred = API.queryFormplayer(options, route);
-        if (options.endpointId) {
-            deferred.fail(function (response) {
+        // If an endpoint is provided, first claim any cases it references, then navigate
+        var user = FormplayerFrontend.getChannel().request('currentUser');
+        var deferred = $.Deferred();
+        $.ajax({
+            type: 'POST',
+            url: initialPageData.reverse('claim_all_cases'),
+            data: {
+                username: user.restoreAs || user.username,
+                case_ids: _.values(options.endpointArgs),
```

Comment:
what about passing using the traditional JQuery serialization here so that we don't get `case_ids[]` and just get `case_ids`?

https://api.jquery.com/jQuery.param/